### PR TITLE
[buteo-sync-plugins] Update helpers for vCard operations

### PR DESCRIPTION
**Pull request by chriadam**
This commit updates the helpers for vCard operations so that they
correctly handle PHOTO fields.  The helpers are copied from the
Seaside (libcontacts) implementation.

### CODE REVIEW

**Comment by chriadam** [comment]
@matthewvogt @hmallat @VDVsx Support photo fields, also update some other things to keep the implementations in sync.

File: storageplugins/hcontacts/ContactPropertyHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,192 @@
+//
+// This implementation is taken from seasidepropertyhandler.cpp in https://github.com/nemomobile/libcontacts
+//
+
+/*
+ * Copyright (C) 2013 Jolla Ltd.
```
Comment (VDVsx) [inline storageplugins/hcontacts/ContactPropertyHandler.cpp]: Update year

File: storageplugins/hcontacts/ContactPropertyHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    const QString propertyName(property.name().toLower());
+
+    if (propertyName == QLatin1String("photo")) {
+        processPhoto(property, alreadyProcessed, updatedDetails);
+    }
+}
+
+static const QString FIELD_SOUND("SOUND");
+static const QString FIELD_SYNCTARGET("X-NEMOMOBILE-ONLINEACCOUNT-DEMO");
+static const QString FIELD_ONLINEACCOUNT("X-NEMOMOBILE-SYNCTARGET");
```
Comment (VDVsx) [inline storageplugins/hcontacts/ContactPropertyHandler.cpp]: QLatin1String ?
These normally appear at top in the file, but also fine for me here :)

Comment (chriadam) [inline storageplugins/hcontacts/ContactPropertyHandler.cpp]: Fixed

File: storageplugins/hcontacts/ContactPropertyHandler.cpp (hunk 1 above)
Comment (chriadam) [inline storageplugins/hcontacts/ContactPropertyHandler.cpp]: Fixed, and in the headers + ContactsImport.cpp

File: storageplugins/hcontacts/ContactsBackend.cpp [hunk 3]
```diff
@@ -409,85 +409,67 @@ QList<QContact> ContactsBackend::convertVCardListToQContactList(const QStringLis
...
-    //CRLF didn't work though.
-    const QString LF = "\n";
-
+    QString crlfSequence;
+    crlfSequence += QChar::CarriageReturn + QChar::LineFeed;
     foreach ( const QString& vcard, aVCardList )
     {
         byteArray.append(vcard.toUtf8());
-        byteArray.append(LF.toUtf8());
+        if (!vcard.endsWith(crlfSequence)) {
```
Comment (matthewvogt) [inline storageplugins/hcontacts/ContactsBackend.cpp]: If the data ends with CR or LF, it should probably be replaced by CRLF.

**Comment by chriadam** [comment]
After review, decided to backport the "applyNameFixes" changes to the Seaside Import implementation, and rewrite the contacts storage plugin here to use SeasideImport.  This will place a new dependency (on libcontacts) here, but will ensure easier maintainability and cleaner code.

**Comment by chriadam** [comment]
Superceded by https://github.com/nemomobile/buteo-sync-plugins/pull/40

**Pull request closed without merging**